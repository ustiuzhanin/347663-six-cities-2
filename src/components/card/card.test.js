import React from 'react';
import renderer from 'react-test-renderer';
import Card from './card.jsx';

test(`Card's snapshot`, () => {
  const tree = renderer.create(
      <Card
        card={{
          id: 1,
          src: `img`,
          price: `10`,
          rating: `10`,
          name: `Beautiful`,
          type: `Private`
        }}
        cardMouseEnterHandler={jest.fn()}
        onCardHeaderClick={jest.fn()}
        cardMouseLeaveHandler={jest.fn()}
      />
  );

  expect(tree).toMatchSnapshot();
});
