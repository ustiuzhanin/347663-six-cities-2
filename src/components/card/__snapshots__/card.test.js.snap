// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card's snapshot 1`] = `
<article
  className="cities__place-card place-card"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  <div
    className="cities__image-wrapper place-card__image-wrapper"
  >
    <a
      href="#"
    >
      <img
        alt="Place image"
        className="place-card__image"
        height="200"
        src="https://htmlacademy-react-2.appspot.com/six-cities/static/hotel/9.jpg"
        width="260"
      />
    </a>
  </div>
  <div
    className="place-card__info"
  >
    <div
      className="place-card__price-wrapper"
    >
      <div
        className="place-card__price"
      >
        <b
          className="place-card__price-value"
        >
          €
          101
        </b>
        <span
          className="place-card__price-text"
        >
          / night
        </span>
      </div>
      <button
        className="place-card__bookmark-button place-card__bookmark-button--active button"
        type="button"
      >
        <svg
          className="place-card__bookmark-icon"
          height="19"
          width="18"
        >
          <use
            xlinkHref="#icon-bookmark"
          />
        </svg>
        <span
          className="visually-hidden"
        >
          In bookmarks
        </span>
      </button>
    </div>
    <div
      className="place-card__rating rating"
    >
      <div
        className="place-card__stars rating__stars"
      >
        <span
          style={
            Object {
              "width": "62%",
            }
          }
        />
        <span
          className="visually-hidden"
        >
          Rating
        </span>
      </div>
    </div>
    <h2
      className="place-card__name"
    >
      <a
        href="/offer/1"
        onClick={[Function]}
      >
        Canal View Prinsengracht
      </a>
    </h2>
    <p
      className="place-card__type"
    >
      room
    </p>
  </div>
</article>
`;
